### DISCUSSION TITLE
How to get setup and working

### DISCUSSION THREAD

**Discussion by ukserversltd** [bootstrap-email/bootstrap-email#155] [Help]
Hi everyone, im struggling to get this setup from scratch can someone advise, ive installed gem and downloaded the code to my folder from github but now stuck, do i j ust place .html file in the root directory of bootstrap-email i got from github and run command bootstrap-email? 

i also. do ruby list and shows bootstrap-email in local but when i run command i get the following:

-bash: bootstrap-email: command not found

If could get some direction on steps to get this setup and compiling would be great, i only have a few days to role out templates for client and stuck. I have paid for bootstrap-email so hoping will be some quick support on this.

Thanks

**Comment by stuyam** [comment]
Happy to help. First off, if you just trying to do something simple, you could give the online editor a try at [app.bootstrapemail.com/editor](https://app.bootstrapemail.com/editor). It has an integration with all the templates so for example on [this page](https://app.bootstrapemail.com/templates/account-subscribed) you can click `Open in Editor` to open the template in the online editor and use it that way to render the HTML.

But if you wanting to use it locally on your computer that is also a great option. Here are the usage instructions to use it via the command line: https://bootstrapemail.com/docs/usage#command-line
You need to have Ruby installed on your computer first, then you should be able to run `gem install bootstrap-email` to install it on your computer. You do NOT need to have the bootstrap-email repo cloned onto your machine for it to work, you can just use via the gem instillation.

If you are still having issues you _could_ clone the bootstrap-email to your computer and in that directory via the cli you can run `bin/bootstrap-email` for the same result for running the CLI.

As for where put the .html files, you can put them in any folder you want and just run `bootstrap-email` in that folder and it will compile all the `.html` files in that folder.

If you are going the cloning route where you cloned bootstrap-email repo and want to compile in a different folder you just need to give the relative path to the `bin/bootstrap-email` file so if you created a folder inside the bootstrap-email folder once you clone then you would need to do `../bin/bootstrap-email`.

Let me know if you need any further assistance. Thanks for buying templates and we will make sure you can run them 👍

**Reply by ukserversltd** [reply]
Thanks for getting back to me, i have ruby installed and installed via gem install but i still get command not found when i run bootstrap-email so i will give it a go using the /bin/ method and let you know but when i did the install before it did not add anythting hence why i downloaded from the repo.

**Reply by ukserversltd** [reply]
When i run the build using /bin path iu get the following:

thor:bootstrap-email-master admin$ bin/bootstrap-email
Compiling file out.html
New css file cached for bootstrap-email
/Users/james/DICE/bootstrap-email-master/lib/bootstrap-email/converters/body.rb:6:in `build': undefined method `[]' for nil:NilClass (NoMethodError)
	from /Users/james/DICE/bootstrap-email-master/lib/bootstrap-email/converters/base.rb:11:in `build'
	from /Users/james/DICE/bootstrap-email-master/lib/bootstrap-email/compiler.rb:59:in `compile_html!'
	from /Users/james/DICE/bootstrap-email-master/lib/bootstrap-email/compiler.rb:23:in `perform_full_compile'
	from /Users/james/DICE/bootstrap-email-master/lib/bootstrap-email/bootstrap_email_cli.rb:77:in `block in <top (required)>'
	from /Users/james/DICE/bootstrap-email-master/lib/bootstrap-email/bootstrap_email_cli.rb:73:in `each'
	from /Users/james/DICE/bootstrap-email-master/lib/bootstrap-email/bootstrap_email_cli.rb:73:in `<top (required)>'
	from bin/bootstrap-email:2:in `require_relative'
	from bin/bootstrap-email:2:in `<main>'
